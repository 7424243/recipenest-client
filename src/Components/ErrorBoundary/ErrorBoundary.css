/* Classes */

.error-boundary-heading {
    padding-top: 50px;
    text-align: center;
    font-family: 'Baskervville', serif;
}