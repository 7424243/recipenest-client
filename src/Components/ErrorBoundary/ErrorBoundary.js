import React, {Component} from 'react'

class ErrorBoundary extends Component {
    constructor(props) {
        super(props)
        this.state = {
            hasError: false
        }
    }

    static getDerivedStateFromError(error) {
        return {hasError: true, error}
    }

    render() {
        if(this.state.hasError) {
            return <h2 className='error-boundary-heading'>Something went wrong, unable to display at this time.</h2>
        } else {
            return this.props.children
        }
    }
}

export default ErrorBoundary