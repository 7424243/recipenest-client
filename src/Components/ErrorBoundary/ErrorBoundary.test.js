import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ErrorBoundary from './ErrorBoundary'


describe('ErrorBoundary Component', () => {

    //create a test function that returns null
    function Something() {
        return null
    }

    //smoke test
    it('renders without crashing if error', () => {
        const div = document.createElement('div')
        ReactDOM.render(<ErrorBoundary><Something /></ErrorBoundary>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ErrorBoundary><Something /></ErrorBoundary>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

})
