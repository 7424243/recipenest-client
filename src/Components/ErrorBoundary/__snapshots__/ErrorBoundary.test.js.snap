// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component renders the UI as expected 1`] = `null`;
